### PR TITLE
Investment Project search created_on filter.

### PR DESCRIPTION
**Pull request by elcct**
This adds `created_on_after` and `created_on_before` filter to the Investment Project search.

It also reorganises tests a little, so that permission tests are in its own place.

### CODE REVIEW

File: datahub/search/investment/test/test_views.py [hunk 1]
```diff
@@ -92,17 +105,47 @@ def test_search_investment_project_json(self, setup_data):
...
+        response = self.api_client.post(url, filter, format='json')
 
         assert response.status_code == status.HTTP_200_OK
-        assert response.data['count'] == 1
-        assert len(response.data['results']) == 1
+        assert response.data['count'] == results
+        assert len(response.data['results']) == results
+
+    @pytest.mark.parametrize('filter,results', (
+        ({'created_on_before': datetime.datetime(2017, 6, 13, 9, 44, 31, 62870)}, 2, ),
```
Comment (reupen) [inline datahub/search/investment/test/test_views.py]: I found this a bit hard to read, maybe make the formatting a bit more consistent? Could also do `import datetime from datetime` to avoid having to write datetime.datetime (although it might make more sense to use strings since they're part of the request – it's not really clear what format they're being serialised to).

File: datahub/search/investment/test/test_views.py [hunk 2]
```diff
@@ -92,17 +105,47 @@ def test_search_investment_project_json(self, setup_data):
         assert len(response.data['results']) == 1
         assert response.data['results'][0]['name'] == 'abc defg'
 
-    def test_search_investment_project_date_json(self, setup_data):
+    @pytest.mark.parametrize('filter,results', (
+        ({'estimated_land_date_before': datetime.datetime(2017, 6, 13, 9, 44, 31, 62870)}, 1, ),
```
Comment (reupen) [inline datahub/search/investment/test/test_views.py]: `estimated_land_date` is a date, any reason for using datetimes?

Might be nice to test some edge cases e.g. when the actual estimated_land_date value is equal to one of filter values.

What happens when you pass a estimated_land_date_after value greater than the estimated_land_date_before value? :)

File: datahub/search/investment/test/test_views.py [hunk 3]
```diff
@@ -92,17 +105,47 @@ def test_search_investment_project_json(self, setup_data):
...
+    ))
+    def test_search_investment_project_created_on_json(self, created_on_data, filter, results):
+        """Tests detailed investment project search."""
+        url = reverse('api-v3:search:investment_project')
+
+        response = self.api_client.post(url, filter, format='json')
+
+        assert response.status_code == status.HTTP_200_OK
+        assert response.data['count'] == results
+        assert len(response.data['results']) == results
```
Comment (reupen) [inline datahub/search/investment/test/test_views.py]: Would be nice to check that they're the expected results (rather than just the count).

File: datahub/search/investment/test/test_views.py [hunk 4]
```diff
@@ -92,17 +105,47 @@ def test_search_investment_project_json(self, setup_data):
...
+        ({'estimated_land_date_after': datetime.datetime(2017, 6, 13, 9, 44, 31, 62870)}, 2,),
+        (
+            {
+                'estimated_land_date_after': datetime.datetime(2017, 6, 13, 9, 44, 31, 62870),
+                'estimated_land_date_before': datetime.datetime(2030, 6, 13, 9, 44, 31, 62870),
+            },
+            1,
+        ),
+    ))
+    def test_search_investment_project_estimated_land_date_json(self, setup_data, filter, results):
```
Comment (marcofucci) [inline datahub/search/investment/test/test_views.py]: Can you use a name different from `filter` so that it doesn't shadow the built-in `filter`

Comment (elcct) [reply datahub/search/investment/test/test_views.py]: Oops. Sure I will fix that, thanks :blush:

File: datahub/search/investment/test/test_views.py (hunk 1 above)
Comment (elcct) [reply datahub/search/investment/test/test_views.py]: I have improved formatting and dropped datetime.datetime in favour of string.

File: datahub/search/investment/test/test_views.py (hunk 2 above)
Comment (elcct) [reply datahub/search/investment/test/test_views.py]: Removed datetimes, added some edge cases.

File: datahub/search/investment/test/test_views.py (hunk 3 above)
Comment (elcct) [reply datahub/search/investment/test/test_views.py]: Also added checking the results. Thanks :)

**Comment by elcct** [comment]
@reupen no particular reason. I have removed it now. Thanks.
